src/index.ts

```diff
@@ -0,0 +1,294 @@
+  files: Record<string, {
+    name: string
+    blob: Blob
+    url?: string
+  }>
+  zip: {
+    name: string
+    blob: Blob
+    url?: string
+  }
+}
+
+export async function SvgPacker({ icons, ...options }: SvgPackerOptions): Promise<SvgPackerResult> {
+  const {
+    fontName = 'iconfont',
+    cssPrefix = 'iconfont',
+    fileName = 'iconfont',
+    startCodepoint = 0xE001,
+    fontHeight = 1000,
+    descent = 0,
+    fixedWidth = false,
+  } = options ?? {}
+
+  const parsedIcons = icons.map(({ svg, name, unicode }, i) => {
+    return {
+      svg,
+      name,
+      unicode: unicode ?? (startCodepoint + i),
+    }
+  })
+
+  const iconStreams = parsedIcons.map(({ svg, unicode, name }) => {
+    const iconStream = new PassThrough()
+    iconStream.write(svg, 'utf8')
+    iconStream.end()
+    iconStream.metadata = {
+      unicode: [String.fromCharCode(unicode)],
+      name,
+    }
+    return iconStream
+  })
+
+  const files = {} as SvgPackerResult['files']
+
+  const zip = await downloadZip(generateZipEntries(
+    iconStreams,
+    {
+      fontName,
+      cssPrefix,
+      fileName,
+      startCodepoint,
+      fontHeight,
+      descent,
+      fixedWidth,
+      icons: parsedIcons,
+    },
+    files,
+  )).blob()
+
+  return {
+    files,
+    zip: {
+      name: `${options.fileName}.zip`,
+      blob: zip,
+      url: makeUrl(zip),
+    },
+  }
+}
+
+function addFile(
+  files: SvgPackerResult['files'],
+  filename: string,
+  ext: string,
+  data: BufferLike,
+  mime = 'text/plain',
+) {
+  const blob = new Blob([data], { type: mime })
+  files[ext] = {
+    name: filename,
+    blob,
+    url: makeUrl(blob),
+  }
+  return {
+    name: filename,
+    input: blob,
+    url: makeUrl(blob),
+  } satisfies InputWithSizeMeta & { url?: string }
+}
+
+function makeUrl(blob: Blob | MediaSource) {
+  if (typeof window === 'undefined' || !window.URL || !window.URL.createObjectURL)
+    return null
+  return window.URL.createObjectURL(blob)
+}
+
+function makeSVG(iconStreams: PassThrough[], options: Omit<SvgPackerOptions, 'icons'>) {
+  return new Promise((resolve) => {
+    const fontStream = new SVGIcons2SVGFontStream(options)
+    const parts = []
+    const decoder = new StringDecoder('utf8')
+    fontStream.on('data', (chunk) => {
+      parts.push(decoder.write(chunk))
+    })
+    fontStream.on('finish', () => {
+      resolve(parts.join(''))
+    })
+    iconStreams.forEach(fontStream.write.bind(fontStream))
+    fontStream.end()
+  })
+}
+
+function makeTTF(svgFont: BufferLike) {
+  const ttfFontBuffer: Uint8Array = svg2ttf(svgFont).buffer
+
+  return ttfFontBuffer
+}
+
+function makeEOT(ttfFontBuffer: Uint8Array) {
+  const eotFontBuffer = ttf2eot(ttfFontBuffer).buffer
+
+  return eotFontBuffer
+}
+
+function makeWOFF(ttfFontBuffer: Uint8Array) {
+  const woffFontBuffer = ttf2woff(new Uint8Array(ttfFontBuffer.buffer)).buffer
+  return woffFontBuffer
+}
+
+async function browserPromise() {
```

Comment:
allow external wasm link, for example with Vite we can use dynamic import to resolve it (maybe adding the wasm url in the options?)

Reply:
this is not required, when using Vite the wasm file will be loaded using `ttf2woff2` that will use the file system